src/functions-and-arrays.js

```diff
@@ -1,73 +1,434 @@
 // Iteration #1: Find the maximum
-
+function maxOfTwoNumbers(num1, num2) {
+  if (num1 > num2) {
+    return num1;
+  } else if (num2 > num1) {
+    return num2;
+  } else {
+    return num1 && num2;
+  }
+}
 // Iteration #2: Find longest word
-const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
-
+//const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
+const words2 = [];
+function findLongestWord(array) {
+  var max = array[0];
+  //max = crocodile
+  if (array.length <= 0) {
+    return null;
+  } else {
+    for (let i = 0; i < array.length; i++) {
+      if (array[i].length > max.length) {
+        max = array[i];
+      }
+    }
+    return max;
+  }
+}
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumNumbers(array) {
+  let sum = 0;
+
+  for (let i = 0; i < array.length; i++) {
+    sum += array[i];
+  }
+  return sum;
+}
+
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
-const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
+//const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
+const numbersAvg = [];
+
+let sum = sumNumbers(numbersAvg);
+
+function averageNumbers(array, sum) {
```

Comment:
It was a really good approach to try to use the previous function result inside this one (to not redo that part) but you were missing a couple of pointers:
- First, by declaring a variable sum outside the function (which in other case would be totally fine) the system is having conflicts with Jasmine, so that is what is breaking your code.
- Also, you can see that the second argument in the function (sum) has the same name as the variable created outside the function. So the function will try to get that argument value when the function is invoked and not the result from the function sumNumbers(numbersAvg).
- A better way to do it would be to remove the variable declaration outside, remove the second argument and just call the function sumNumbers(array) inside the function averageNumbers and use the result.